core/controllers/editor.py

```diff
@@ -116,33 +186,21 @@ def get(self, exploration_id):
     def put(self, exploration_id):
         """Updates properties of the given exploration."""
         exploration = exp_fetchers.get_exploration_by_id(exploration_id)
-        version = self.payload.get('version')
-        if version is None:
-            raise base.BaseHandler.InvalidInputException(
-                'Invalid POST request: a version must be specified.')
+        version = self.normalized_payload.get('version')
+
         if version > exploration.version:
             raise base.BaseHandler.InvalidInputException(
                 'Trying to update version %s of exploration from version %s, '
                 'which is not possible. Please reload the page and try again.'
                 % (exploration.version, version))
 
-        commit_message = self.payload.get('commit_message')
-
-        if (commit_message is not None and
-                len(commit_message) > constants.MAX_COMMIT_MESSAGE_LENGTH):
-            raise self.InvalidInputException(
-                'Commit messages must be at most %s characters long.'
-                % constants.MAX_COMMIT_MESSAGE_LENGTH)
-
-        change_list_dict = self.payload.get('change_list')
+        commit_message = self.normalized_payload.get('commit_message')
+        change_list_dict = self.normalized_payload.get('change_list')
```

Comment:
I am planning to create a seperate PR by 15th August, in which I will do the requested changes, since it will help me to segregate my PR's because this one is intended to add schemas.
The PR I am going to create is intended to modify architecture.